src/frontend/query_commands.ml

```diff
@@ -274,6 +274,49 @@ let dispatch pipeline (type a) : a Query_protocol.t -> a =
     ignore (Type_utils.type_in_env ~verbosity ~context env ppf source : bool);
     to_string ()
 
+  | Stack_or_heap_enclosing (pos, index) ->
```

Comment:
Is it easy to share code between this case and the type-enclosing case? If so, it seems a bit better for merging to factor out the code. (That way, if we pick up new changes to `Type_enclosing` in a rebase, we will be handed a merge conflict instead of silently applying the edit to one case and dropping it from the other case.)

Reply:
But this isn't that strong of a concern, so if the code is easier to structure by having separate cases, then I'm in favor of leaving it as-is.

Reply:
I took a stab at this. It turns out there's not that much shared logic, but I think abstracting out some of the more low-level details anyways made the code cleaner, so I'm in favor of keeping the change.

Reply:
I find the new code harder to read, as you need to cross-reference two places to understand what's going on. I bet my original suggestion was just bad. A lighter weight way to achieve my suggestion may be just leaving a comment saying that the two code blocks should stay in sync.